src/solvers/quasi_Newton.jl

```diff
@@ -284,24 +289,24 @@ end
 
 function step_solver!(mp::AbstractManoptProblem, qns::QuasiNewtonState, iter)
     M = get_manifold(mp)
-    qns.X = get_gradient(mp, qns.p)
+    get_gradient!(mp, qns.X, qns.p)
     η = qns.direction_update(mp, qns)
     α = qns.stepsize(mp, qns, iter, η)
-    p_old = copy(M, get_iterate(qns))
+    copyto!(M, qns.p_old, get_iterate(qns))
     αη = α * η
     retract!(M, qns.p, qns.p, η, α, qns.retraction_method)
     β = locking_condition_scale(
-        M, qns.direction_update, p_old, αη, qns.p, qns.vector_transport_method
+        M, qns.direction_update, qns.p_old, αη, qns.p, qns.vector_transport_method
     )
     vector_transport_to!(
-        M, qns.sk, p_old, αη, qns.p, get_update_vector_transport(qns.direction_update)
+        M, qns.sk, qns.p_old, αη, qns.p, get_update_vector_transport(qns.direction_update)
     )
     vector_transport_to!(
-        M, qns.X, p_old, qns.X, qns.p, get_update_vector_transport(qns.direction_update)
+        M, qns.X, qns.p_old, qns.X, qns.p, get_update_vector_transport(qns.direction_update)
     )
-    new_grad = get_gradient(mp, qns.p)
-    qns.yk .= new_grad ./ β .- qns.X
-    update_hessian!(qns.direction_update, mp, qns, p_old, iter)
+    get_gradient!(mp, qns.X_old, qns.p)
```

Comment:
hm, this now looks like you use `X_old` for a new gradient? could we then maybe switch the roles of X and X_old? This way it is confusing, I did not just mean to call new_grad->X_old

Reply:
could you check for this naming? Currently it might really to be too confusing, since we use a `X_old` field for a new gradient.

Reply:
I think I might have misunderstood you, you just want to switch `X` and `X_old`, right? I can do that.

Reply:
But then be careful when someone accesses `get_gradient` what to return ;)

Reply:
Then maybe it would be easier to change `X_old` to `X_new`?

Reply:
That would be a bit inconsistent to what we do in other solvers, since we usually only store `X_old` and not `X_new`, but if you do not see how to best rephrase this, then ok take `X_new`.

Reply:
I've changed usage of `X_old` and `X` a bit and I think it's somewhat consistent now. What do you think?